make/template/main.mk

```diff
@@ -74,7 +74,10 @@ else
 endif
 
 ifneq ($(COMPILER), ICC)
-  CORECXXFLAGS += -Woverloaded-virtual -Wshadow
+    CORECXXFLAGS += -Woverloaded-virtual -Wshadow -Werror
```

Comment:
I don't think we want -Werror left here, @SadieCat ? Debug builds and CI enable that.

Reply:
Ah, yeah. -Werror is enabled in debug builds already. Enabling it in normal builds is a bit excessive.

Reply:
I left this in by mistake as it was helpful for me to catch issues. I think it could be enabled to prevent backsliding, but having it in debug is probably enough.